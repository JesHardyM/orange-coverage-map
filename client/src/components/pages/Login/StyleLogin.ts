import styled from "styled-components";

const StyledLogin = styled.div`

  h1{
    font-family: 'HelvNeue75_W1G', sans-serif;;
  }

  form{
    display: flex;
    flex-direction: column;
    gap: 1rem;
  }
  

  .button{
    width: 5rem;
  }
`;

export default StyledLogin;
