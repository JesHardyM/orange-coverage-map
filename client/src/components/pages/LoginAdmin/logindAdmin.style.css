.black {
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.button__login{
    width: 17rem;
}

h4 {
    color: white
}

form{
    margin-bottom: 8rem;
}