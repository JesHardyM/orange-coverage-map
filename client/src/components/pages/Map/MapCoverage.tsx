
const MapCoverage = () => {
  return (
    <>
    <div className="container py-4 px-3 mx-auto d-flex justify-content-center" style={{height: '85vh'}}>
      <div className="row">
      <div className="col-6 d-flex justify-content-center align-items-center">
        <img src="src/assets/images/map-coverage.svg" className="img-thumbnail" alt="map-coverage"></img>
        </div>

        <div className="col-6 d-flex justify-content-center align-items-center flex-wrap flex-column gap-2">
        <button type="button" className="btn btn-primary">Primary</button>
        <button type="button" className="btn btn-primary">Primary</button>
        <button type="button" className="btn btn-primary">Primary</button>
        <button type="button" className="btn btn-primary">Primary</button>
        <button type="button" className="btn btn-primary">Primary</button>
        </div>
        </div>
        </div>
    </>
  )
}

export default MapCoverage;