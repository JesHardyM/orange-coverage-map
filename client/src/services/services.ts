import { FieldValues } from "react-hook-form";
import {
  CompaniesInterface,
  ListInterface,
  ServicesInterface,
  adminsEndpoint,
  companiesEndPoint,
  dataListEndpoint,
  networkQualityEndpoint,
  registerEndPoint,
} from "./service.module";

export class Services implements ServicesInterface {
  constructor() {}

  //✅
  async getDataList(): Promise<{ response: ListInterface[] } | undefined> {
    try {
      const data = await fetch(dataListEndpoint);
      const response: ListInterface[] = await data.json();
      console.log(response)
      return { response };
    } catch (error: unknown) {
      if (error instanceof Error) {
        throw new Error(error.message);
      }
    }
  }

  //✅
  async getCompanies(): Promise<{ response: CompaniesInterface[] } | undefined> {
    try {
      const data = await fetch(companiesEndPoint);
      const response: CompaniesInterface[] = await data.json();
      console.log(response)
      return { response };
    } catch (error: unknown) {
      if (error instanceof Error) {
        throw new Error(error.message);
      }
    }
  }

  //✅
  async postDataList(
    body: FieldValues
  ): Promise<ListInterface | undefined | boolean> {
    try {
      const data = await fetch(networkQualityEndpoint, {
        method: "POST",
        headers: { "content-type": "application/json" },
        body: JSON.stringify(body),
      });
      const response: ListInterface = await data.json();
      
      if (response) {
        return response;
      } else {
        return false
      }
    } catch (error: unknown) {
      if (error instanceof Error) {
      throw new Error(error.message);
    }
    }
  }

    //✅
    async postLoginAdmin(
      body: FieldValues
    ): Promise<FieldValues | undefined | boolean> {
      try {
        const data = await fetch(adminsEndpoint, {
          method: "POST",
          headers: { "content-type": "application/json" },
          body: JSON.stringify(body),
        });
        const response = await data.json();
  
        localStorage.setItem("token", response.token);
  
        if (response) {
          return true;
        } else {
          return false;
        }
      } catch (error) {
        if (error instanceof Error) {
          throw new Error(error.message);
        }
      }
    }

  //✅
  async postRegisterUser(body: FieldValues): Promise<FieldValues | undefined> {
    try {
      const data = await fetch(registerEndPoint, {
        method: "POST",
        headers: { "content-type": "application/json" },
        body: JSON.stringify(body),
      });
      const response: FieldValues = await data.json();
      return response;
    } catch (error) {
      if (error instanceof Error) {
        throw new Error(error.message);
      }
    }
  }
  
}

export const services = new Services();
