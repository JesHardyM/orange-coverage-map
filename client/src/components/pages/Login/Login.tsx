import { useForm } from "react-hook-form";
import { services } from "../../../services";
import { Link, useNavigate } from "react-router-dom";

interface FormData {
  Email: string;
  ContraseÃ±a: string;
}

const Login = () => {
  const { register, handleSubmit } = useForm<FormData>();
  const navigate = useNavigate();

  const postData = (data: FormData) => {
    console.log(data);

    const response: any = services.postData("http", data);

    if (response.token) {
      localStorage.setItem("token", response.token);
      console.log("Successful Authentication");
      navigate("/Home");
      console.log(null);
    } else {
      console.log("invelid credencial");
    }
  };


  return (
    <>
      <div className="d-flex flex-column align-items-center justify-content-center" style={{ height: '70vh' }}>

      <div className="d-flex align-items-center justify-content-center h-100">
      <img src="src/assets/icons/register-icon.svg" alt="register-icon" className="m-3" style={{ width: '70%', height: '80%' }}/>
      </div>

      <Link type="button" className="btn btn-primary" to={'/register'} >Register</Link>

      </div>
    </>
  );
};

export default Login;
