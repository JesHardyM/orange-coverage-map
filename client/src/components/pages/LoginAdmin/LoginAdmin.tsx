import { FieldValues, useForm } from "react-hook-form";
import HeaderLoginAdmin from "../../templates/HeaderLoginAdmin/HeaderLoginAdmin";
import React from "react";
import { services } from "../../../services/services";
import ButtonOrange from "../../atoms/ButtonOrange";
import './logindAdmin.style.css'

const LoginAdmin = (): React.JSX.Element => {
  const {
    register,
    formState: { errors },
    handleSubmit,
  } = useForm();

  const postLogin = (data: FieldValues) => {
    services.postLoginAdmin(data);
  };

  return (
    <>
      <HeaderLoginAdmin />
      <div className="black">
      <div className="container py-4 px-3 mx-auto ">
       
        <h4>Identificate</h4>

        <form className="" action="" onSubmit={handleSubmit(postLogin)}>
        <div>
          <div className="mb-3">
            <label htmlFor="input__name" className="form-label">
              Usuario
            </label>
            <input
              type="text"
              className="form-control"
              id="input__name"
              placeholder="Usuario"
              {...register("admin_name", {
                required: true,
              })}
            />
            {errors.admin_name?.type === "required" && (
              <p className="text-danger fw-bold">name required</p>
            )}
          </div>

          <div className="mb-3">
            <label htmlFor="input__password" className="form-label">
              Contraseña
            </label>
            <input
              type="password"
              className="form-control"
              id="input__password"
              placeholder="Contraseña"
              {...register("admin_password", {
                required: true,
              })}
            />
            {errors.admin_password?.type === "required" && (
              <p className="text-danger fw-bold">password required</p>
            )}
          </div>
          </div>
          <div className="text-center w-100">
          <ButtonOrange textButton="Entrar" direction="/map-coverage"/>
          </div>
        </form>
      </div>
      </div>
    </>
  );
};

export default LoginAdmin;
