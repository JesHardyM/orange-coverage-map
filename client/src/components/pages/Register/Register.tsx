import { FieldValues, useForm } from "react-hook-form";
import Header from "../../templates/Header/Header";
import { useEffect, useState } from "react";
import { Link, useNavigate } from "react-router-dom";
import { useLoaderData } from "react-router-dom";
import { services } from "../../../services/services";
import Modal from "../../templates/Modal/Modal";
import "./register.style.css";
import { CompaniesInterface } from "../../../services/service.module";
import { createCookie } from "./register.module";


const Register = () => {
  const [change, setChange] = useState(false);
  const [showErrorModal, setShowErrorModal] = useState(false);
  const [errorMessage, setErrorMessage] = useState("");
  const [formData, setFormData] = useState<FieldValues>({});
  const {
    register,
    formState: { errors },
    handleSubmit,
    setValue,
    watch
  } = useForm();

  const navigate = useNavigate();





  useEffect(() => {
    const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
    
    const isiOS = /iPad|iPhone|iPod/.test(navigator.userAgent);
    console.log(isSafari);
    console.log(isiOS);

    console.log(navigator.userAgent)


    // if (isiOS && isSafari) {
    //   navigate('/blocking');
    // }
    
  }, [navigate]);

  const { response } = useLoaderData() as { response: CompaniesInterface[] };

  // Observar los cambios en los campos del formulario
  const formValues = watch();

  // Efecto para almacenar los valores del formulario en localStorage cada vez que cambien
  useEffect(() => {
    for (const [key, value] of Object.entries(formValues)) {
      localStorage.setItem(key, value);
    }
  }, [formValues]);

  // Efecto para rellenar el formulario con valores de localStorage cuando el componente se monta
  useEffect(() => {
    const formFields = ["user_name", "user_lastname", "company_id", "postal_code", "user_check"];
    formFields.forEach(field => {
      const savedValue = localStorage.getItem(field);
      if (savedValue !== null) {
        setValue(field, savedValue);
      }
    });
  }, [setValue]);

  const handleClose = () => {
    setChange(false);
  };

  const handleCloseErrorModal = () => {
    setShowErrorModal(false);
  };

  useEffect(()=>{
    const galleta = document.cookie
    console.log(galleta)

    if (galleta) {
      navigate('/send-data')
    }
  },[navigate])


  const postRegister = async (data: FieldValues) => {
    
    try {
      const response: FieldValues | undefined = await services.postRegisterUser(
        data
      );



      const userUuid = response?.user_id; // Asumiendo que el servidor devuelve el UUID como user_id
      console.log("UUID received from server:", userUuid);

      console.log("UUID received from server:", userUuid);
      if (userUuid) {
        createCookie('userId', userUuid, 365); 
        navigate('/send-data'); 
      } else {
        console.error("No UUID present in the response");
      }

      // Almacenar el UUID en el almacenamiento local
      // localStorage.setItem('userUuid', userUuid);

    } catch (error) {
      // Aquí manejamos el error de registro
      let message =
        "There was an unexpected error during registration. Please try again later.";

      if (error instanceof Error) {
        if (error.response?.status === 409) {
          message =
            "An account with the provided details already exists. Please log in or use different details.";
        } else if (error.message) {
          message += ` Error: ${error.message}`;
        }
      }
      setErrorMessage(message);
      setShowErrorModal(true);
    }
  };

  // Efecto para cargar datos del formulario al montar el componente
  useEffect(() => {
    const storedFormData = localStorage.getItem("formData");
    if (storedFormData) {
      setFormData(JSON.parse(storedFormData));
    }
  }, []);

  // Efecto para guardar datos del formulario en localStorage
  useEffect(() => {
    localStorage.setItem("formData", JSON.stringify(formData));
  }, [formData]);

  return (
    <>
      <Header title="Registro" />
      <div className="container py-4 px-3 d-flex flex-column mt-4 container-form">
        <form className="row g-3" onSubmit={handleSubmit(postRegister)}>
          <div className="col-md-6">
            <label htmlFor="input_name" className="form-label">
              Nombres
            </label>
            <input
              type="text"
              className="form-control"
              id="input_name"
              data-testid="input_name"
              {...register("user_name", {
                required: true,
                pattern:
                  /^([A-ZÁÉÍÓÚÜÑ]?[a-záéíóúüñÁÉÍÓÚÜÑ']+)?(\s[A-ZÁÉÍÓÚÜÑ]?[a-záéíóúüñÁÉÍÓÚÜÑ']+)?$/,
              })}
            />
            {errors.user_name?.type === "required" && (
              <p className="text-danger fw-bold">nombre requerido</p>
            )}
            {errors.user_name?.type === "pattern" && (
              <p className="text-danger fw-bold">nombre invalido</p>
            )}
          </div>
          <div className="col-md-6">
            <label htmlFor="input_last-name" className="form-label">
              Apellidos
            </label>
            <input
              type="text"
              className="form-control"
              id="input_last-name"
              data-testid="input_last-name"
              {...register("user_lastname", {
                required: true,
                pattern:
                  /^([A-ZÁÉÍÓÚÜÑ]?[a-záéíóúüñÁÉÍÓÚÜÑ']+)?(\s[A-ZÁÉÍÓÚÜÑ]?[a-záéíóúüñÁÉÍÓÚÜÑ']+)?$/,
              })}
            />
            {errors.user_lastname?.type === "required" && (
              <p className="text-danger fw-bold">apellidos requerido</p>
            )}
            {errors.user_lastname?.type === "pattern" && (
              <p className="text-danger fw-bold">apellidos invalido</p>
            )}
          </div>
          <div className="col-md-4">
            <label htmlFor="company_select" className="form-label">
              Compañia
            </label>
            <select
              id="company_select"
              className="form-select"
              {...register("company_id", { required: true })}
            >
              <option value="" disabled>
                elige una compañia
              </option>
              {response.map((company) => (
                <option key={company.company_id} value={company.company_id}>
                  {company.company_name}
                </option>
              ))}
            </select>

            {errors.company_id?.type === "required" && (
              <p className="text-danger fw-bold">compañia requerida</p>
            )}
          </div>

          <div className="col-md-2">
            <label htmlFor="input_postal-code" className="form-label">
              Codigo Postal
            </label>
            <input
              type="text"
              className="form-control"
              data-testid="input_postal-code"
              id="input_postal-code"
              {...register("postal_code", {
                required: true,
                pattern: /^\d{5}$/,
              })}
            />
            {errors.postal_code?.type === "required" && (
              <p className="text-danger fw-bold">codigo postal requerido</p>
            )}
            {errors.postal_code?.type === "pattern" && (
              <p className="text-danger fw-bold">codigo postal invalido</p>
            )}
          </div>

          <div className="col-12">
            <div className="form-check">
              <input
                className="form-check-input"
                type="checkbox"
                id="input_check"
                data-testid="input_check"
                {...register("user_check", {
                  required: true,
                })}
              />
              <label className="form-check-label" htmlFor="input_check">
                Estoy de acuerdo con la{" "}
                <Link
                  to={"/terms-conditions"}
                >
                  política de privacidad
                </Link>
              </label>
              {errors.user_check?.type === "required" && (
                <p className="text-danger fw-bold">
                  debe aceptar los terminos y condiciones
                </p>
              )}
            </div>
          </div>

          <div className="button-register d-flex justify-content-center w-100">
            <button
              type="submit"
              className="btn btn-primary mt-2 button-submit"
            >
              Me Registro
            </button>
          </div>
        </form>
      </div>

      <Modal
        to={"/send-data"}
        button={"Continuar"}
        display={change}
        onClose={handleClose}
        modalTitle={"¡Registro con éxito!"}
      />

      {/* Error Modal */}
      <Modal
        button={"Close"}
        display={showErrorModal}
        onClose={handleCloseErrorModal}
        modalTitle={errorMessage}
        to=""
      />
    </>
  );
};

export default Register;
