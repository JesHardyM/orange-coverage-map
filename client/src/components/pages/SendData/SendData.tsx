import { useEffect, useState } from "react";
import Header from "../../templates/Header/Header";
import { useNetwork } from "../../../hooks/useNetwork";
import { services } from "../../../services/services";
import ModalSucces from "../../templates/ModalSucces/ModalSucces";
import ModalError from "../../templates/ModalError/ModalError";
import "./sendData.style.css";
import { useNavigate } from "react-router-dom";
import Modal from "../../templates/Modal/Modal";


const SendData = (): React.JSX.Element => {
  const [ModalErr, setModalError] = useState(false);
  const [ModalSuccess, setModalSuccess] = useState(false);
  const [Modall, setModall] = useState(false)
  const networkInfo = useNetwork(null);
  const userUuid = localStorage.getItem("userUuid"); // Recuperamos el UUID del localStorage

  const navigate = useNavigate();


  const closeModal =()=>{
    setModall(false)
  }

  useEffect(()=>{
    if (typeof window.InstallTrigger !== "undefined") {
      navigate("/blocking");
    } else if (typeof window.safari !== "undefined") {
      navigate("/blocking");
    }


      const findCookie = document.cookie;
      console.log(findCookie);
  
      if (!findCookie) {
        navigate("/login");
      }

  },[navigate])


  

  const sendLocation = (): void => {
    navigator.geolocation.getCurrentPosition(
      async (position) => {
        try {
          const geoLocationData = {
            latitude: position.coords.latitude,
            longitude: position.coords.longitude,
          };

          const combinedData = {
            ...geoLocationData,
            ...networkInfo,
            network: networkInfo.effectiveType,
            userUuid, // Enviamos el UUID junto con los datos de la red
          };

          console.log(combinedData);

          const postSuccess = await services.postDataList(combinedData);

          console.log(postSuccess)

          if (postSuccess) {

            setModalSuccess(true);

            setTimeout(()=>{
              setModall(true)
            },4000)

            setTimeout(() => {
              setModalSuccess(false);
            }, 3000);

          } 
        } catch (error) {
          console.error("Error al enviar los datos de red:", error);
          if (error) {
            setModalError(true);
              setTimeout(() => {
                setModalError(false);
              }, 3000);
          }
        }
      },
      (error) => {

        navigator.geolocation.watchPosition(
          () => {
            console.log("compartiendo ubicacion");
          },
          () => {
            navigate("/permission");
          }
        );

        console.error("Error al obtener la geolocalización:", error);
      }
    );
  };

  return (
    <>
      <Header title="Enviar Datos" />
      <div className="container py-4 px-3 mx-auto b-1 text-center mt-3">
        <div className="d-flex flex-column align-items-center justify-content-center container-sendData">
          <img src="src/assets/images/send-data.svg" alt="send-data" />
          <p className="w-80">
            Comparte tu calidad de red junto a tu ubicación.
          </p>
          <button
            type="button"
            className="btn btn-primary button__send-data"
            onClick={sendLocation}
          >
            Enviar
          </button>
        </div>
        <ModalSucces display={ModalSuccess}/>
        <ModalError display={ModalErr}/>
        <Modal  display={Modall} buttonText="Aceptar"
       modalTitle="Obtuviste 1 crédito" onClose={closeModal}/>
      </div>
    </>
  );
};

export default SendData;
