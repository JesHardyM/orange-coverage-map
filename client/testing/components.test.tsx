import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { vi } from "vitest";
import Blocking from "../src/components/pages/Blocking/Blocking";
import Condicions from "../src/components/pages/Condicions/Condicions";
import Permission from "../src/components/pages/Permission/Permission";
import Login from "../src/components/pages/Login/Login";
import LoginAdmin from "../src/components/pages/LoginAdmin/LoginAdmin";
import Register from "../src/components/pages/Register/Register";
import SendData from "../src/components/pages/SendData/SendData";
import MapCoverage from "../src/components/pages/Map/MapCoverage";
import { Services } from "../src/services/services";

describe("rendering pages", () => {
  describe("blocking", () => {
    test("image", () => {
      vi.mock("react-router-dom", () => {
        const originalModule = vi.importActual("react-router-dom");
        return {
          ...originalModule,
          Link: ({ to }: { to: string }) => <a href={to}></a>,
        };
      });
      render(<Blocking />);
      const imageBlocking = screen.getByAltText(/blocking-image/i);

      expect(imageBlocking).toHaveAttribute(
        "src",
        "src/assets/images/blocking.svg"
      );
    });

    test("text", () => {
      render(<Blocking />);

      const text = screen.getByTestId(/text/i);
      expect(text).toBeInTheDocument();
    });

    test("button", () => {
      render(<Blocking />);

      const linkElement = document.querySelector('[href="/login"]');
      expect(linkElement).toBeInTheDocument();
    });
  });

  describe("Condicions", () => {


    test("title", () => {
      render(<Condicions />);
      const textElement = screen.getByTestId("text-condicions");
      expect(textElement).toBeInTheDocument();
    });

    test("list", () => {
      render(<Condicions />);
      const listElement = screen.getByTestId("list-condicions");
      expect(listElement).toBeInTheDocument();
    });
  });

  describe("permission", () => {
    test("image", () => {
      render(<Permission />);
      const imagePemission = screen.getByAltText(/permission-image/i);

      expect(imagePemission).toHaveAttribute(
        "src",
        "src/assets/images/blocking.svg"
      );
    });
  });

  describe("login user", () => {
    test("image", () => {
      render(<Login />);
      const imageLogin = screen.getByAltText(/login-image/i);

      expect(imageLogin).toHaveAttribute("src", "src/assets/images/login.svg");
    });
  });

  describe("loginAdmin", () => {
    vi.mock("react-router-dom", async () => {
      return {
        Link: ({ to }: { to: string }) => <a href={to}></a>,
        useNavigate: vi.fn(),
      };
    });

    test("render", () => {
      const { asFragment } = render(<LoginAdmin />);
      expect(asFragment()).toMatchSnapshot();
    });

    describe("Register", () => {
      vi.mock("react-router-dom", async () => {
        return {
          useNavigate: vi.fn(),
          useLoaderData: () => ({
            response: [
              {
                company_name: "Orange",
                company_id: 1,
              },
            ],
          }),
          Link: ({ to }: { to: string }) => <a href={to}></a>,
        };
      });

      test("register", () => {
        const { asFragment } = render(<Register />);
        expect(asFragment()).toMatchSnapshot();
      });
    });



    describe("Map", () => {
      test("map-coverage", () => {
        const { asFragment } = render(<MapCoverage />);
        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  //   describe("services", () => {
  //     let services: Services | null = null;
  //     beforeAll(() => {
  //       services = new Services()
  //     });

  //     test("getDataList", async () => {
  //       const response = await services?.getDataList();
  //       expect(response?.response[0]).toStrictEqual({
  //         created_at: "2023-11-23T11:36:28.000Z",
  //         downlink: 10,
  //         id: 1,
  //         latitude: 40.3632,
  //         longitude: -3.59222,
  //         network: "4g",
  //         rtt: 50,
  //         user_uuid: "2e35bbcc-878d-4740-843d-7583ee47096b",
  //       });
  //     });

  //     test("getCompanies", async () => {
  //       const response = await services?.getCompanies();
  //       expect(response?.response[0]).toStrictEqual({
  //         company_id: 1,
  //         company_name: "Jazztel",
  //       });
  //     });

  //     test("postDataList", async () => {
  //       const response = await services?.postDataList({
  //         created_at: "2023-11-23T11:36:28.000Z",
  //         downlink: 10,
  //         latitude: 40.3632,
  //         longitude: -3.59222,
  //         network: "4g",
  //         rtt: 50,
  //         user_uuid: "2e35bbcc-878d-4740-843d-7583ee47096b",
  //       });
  //       expect(response).toBe(response);
  //     });

  //     test("postLoginAdmin", async () => {
  //       const response = await services?.postLoginAdmin({
  //         admin_username: "Bryan",
  //         admin_password: "1234567",
  //       });
  //       expect(response).toBe(true);
  //     });

  //     test("postRegisterUser", async () => {
  //       const response = await services?.postRegisterUser({
  //         user_name: "usuario",
  //         user_lastname: "last",
  //         company_id: "Orange",
  //         postal_code: 28028,
  //         user_check: true,
  //       });
  //       expect(response).toBe(response);
  //     });
  //   });
  // });
});













vi.mock("../src/services/services.ts", () => {
  return {
    Services: class MockServices {
      constructor() {}

      async getDataList(someCondition: boolean) {
        if (someCondition) {
          throw new Error("error");
        } else {
          return {
            response: [
              {
                created_at: "2023-11-23T11:36:28.000Z",
                downlink: 10,
                id: 1,
                latitude: 40.3632,
                longitude: -3.59222,
                network: "4g",
                rtt: 50,
                user_uuid: "2e35bbcc-878d-4740-843d-7583ee47096b",
              },
            ],
          };
        }
      }
      async getCompanies(someCondition: boolean) {
        if (someCondition) {
          throw new Error("error");
        } else {
          return {
            response: [
              {
                company_id: 1,
                company_name: "Jazztel",
              },
            ],
          };
        }
      }
      async postDataList(someCondition: boolean) {
        if (someCondition) {
          throw new Error("error");
        } else {
          return {
            response: [
              {
              created_at: "2023-11-23T11:36:28.000Z",
              downlink: 10,
              latitude: 40.3632,
              longitude: -3.59222,
              network: "4g",
              rtt: 50,
              user_uuid: "2e35bbcc-878d-4740-843d-7583ee47096b",
              },
            ],
          };
        }
      }
      async postLoginAdmin(someCondition: boolean) {
        if (someCondition) {
          throw new Error("error");
        } else {
          return {
            response: [
              {
                admin_name: "Usuario",
                admin_password: "1234567",
              },
            ],
          };
        }
      }
      async postRegisterUser(someCondition: boolean) {
        if (someCondition) {
          throw new Error("error");
        } else {
          return {
            response: [
              {
          user_name: "usuario",
          user_lastname: "last",
          company_id: "Orange",
          postal_code: 28028,
          user_check: true,
              },
            ],
          };
        }
      }
    },
  };
});






describe("getDataList", () => {
  const nuevo = new Services();

  test("success", async () => {
    const success = await nuevo.getDataList(false);
    expect(success?.response.length).toBe(1);
  });

  test("error", async () => {
    try {
      const errores = await nuevo.getDataList(true);
      expect(errores).toBeFalsy();
    } catch (error) {
      if (error instanceof Error) {
        expect(error.message).toBe("error");
      }
    }
  });
});

describe("getCompanies", () => {
  const services = new Services();

  test("success", async () => {
    const success = await services.getCompanies(false);
    expect(success?.response.length).toBe(1);
  });

  test("error", async () => {
    try {
      const errores = await services.getCompanies(true);
      expect(errores).toBeFalsy();
    } catch (error) {
      if (error instanceof Error) {
        expect(error.message).toBe("error");
      }
    }
  });
});





// describe("postDataList", () => {
//   const services = new Services();

//   test("success", async () => {
//     const success = await services.postDataList(false);
//     expect(success).toMatchObject({
//       created_at: "2023-11-23T11:36:28.000Z",
//       downlink: 10,
//       latitude: 40.3632,
//       longitude: -3.59222,
//       network: "4g",
//       rtt: 50,
//       user_uuid: "2e35bbcc-878d-4740-843d-7583ee47096b",
//     });
//   });

//   test("error", async () => {
//     try {

//       const errores = await services.postDataList(true);
//       expect(errores).toBeFalsy();
//     } catch (error) {
//       if (error instanceof Error) {
//         expect(error.message).toBe("error");
//       }
//     }
//   });
// });
