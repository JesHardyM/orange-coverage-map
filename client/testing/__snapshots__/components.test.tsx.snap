// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rendering pages > loginAdmin > Map > map-coverage 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
    data-testid="headerAdmin"
  >
    <div
      class="container-fluid"
    >
      <div
        class="navbar-brand"
      >
        <div
          class="stretched-link"
        >
          <img
            alt="Boosted - Back to Home"
            height="50"
            loading="lazy"
            src="src/assets/images/logo-orange.svg"
            width="50"
          />
        </div>
      </div>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarText"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse align-items-end"
        id="navbarText"
      >
        <ul
          class="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              href="/map-coverage"
            />
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/data-list"
            />
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/login-admin"
            />
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    class="black"
  >
    <div
      class="container py-4 px-3 mx-auto container__map"
    >
      <div
        class="row"
      >
        <div
          class="col-12 d-flex justify-content-center align-items-center flex-column"
        >
          <div>
            ...cargando
          </div>
        </div>
        <div
          class="col-12 d-flex justify-content-center align-content-center flex-wrap gap-2 mt-3"
        >
          <button
            class="btn rounded "
            id="button4g"
          >
            4G
            <img
              alt="button4g"
              src="src/assets/icons/4g.svg"
            />
          </button>
          <button
            class="btn rounded"
            id="button3g"
          >
            3G
            <img
              alt="button3g"
              src="src/assets/icons/3g.svg"
            />
          </button>
          <button
            class="btn rounded"
            id="button2g"
          >
            2G
            <img
              alt="button2g"
              src="src/assets/icons/2g.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering pages > loginAdmin > Register > register 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="stretched-link"
            href="#"
          >
            <img
              alt="Boosted - Back to Home"
              height="50"
              loading="lazy"
              src="src/assets/images/logo-orange.svg"
              width="50"
            />
          </a>
          <h1
            class="title"
          >
            Registro
          </h1>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="container py-4 px-3 d-flex flex-column mt-4 container-form"
  >
    <form
      class="row g-3"
    >
      <div
        class="col-md-6"
      >
        <label
          class="form-label"
          for="input_name"
        >
          Nombre
        </label>
        <input
          class="form-control"
          data-testid="input_name"
          id="input_name"
          name="user_name"
          type="text"
        />
      </div>
      <div
        class="col-md-6"
      >
        <label
          class="form-label"
          for="input_last-name"
        >
          Apellidos
        </label>
        <input
          class="form-control"
          data-testid="input_last-name"
          id="input_last-name"
          name="user_lastname"
          type="text"
        />
      </div>
      <div
        class="col-md-4"
      >
        <label
          class="form-label"
          for="company_select"
        >
          Compañía
        </label>
        <select
          class="form-select"
          id="company_select"
          name="company_id"
        >
          <option
            disabled=""
            value=""
          >
            Elige una compañía
          </option>
          <option
            value="1"
          >
            Orange
          </option>
        </select>
      </div>
      <div
        class="col-md-2"
      >
        <label
          class="form-label"
          for="input_postal-code"
        >
          Código Postal
        </label>
        <input
          class="form-control"
          data-testid="input_postal-code"
          id="input_postal-code"
          name="postal_code"
          type="text"
        />
      </div>
      <div
        class="col-12"
      >
        <div
          class="form-check"
        >
          <input
            class="form-check-input"
            data-testid="input_check"
            id="input_check"
            name="user_check"
            type="checkbox"
          />
          <label
            class="form-check-label"
            for="input_check"
          >
            Estoy de acuerdo con la 
            <a
              href="/terms-conditions"
            />
          </label>
        </div>
      </div>
      <div
        class="button-register d-flex justify-content-center w-100"
      >
        <button
          class="btn btn-primary mt-2 button-submit"
          type="submit"
        >
          Me registro
        </button>
      </div>
    </form>
  </div>
  <div
    class="modal d-none d-flex  align-items-center h-100 container-modal"
  >
    <div
      class="modal-dialog w-100 "
    >
      <div
        class="container-content modal-content d-flex justify-content-center align-items-center gap-4 "
      >
        <div
          class="modal-header d-flex justify-content-center"
        >
          <h5
            class="modal-title fs-5"
          >
            ¡Registro con éxito!
          </h5>
        </div>
        <div
          class="modal-footer"
        >
          <a
            href="/send-data"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal d-none d-flex  align-items-center h-100 container-modal"
  >
    <div
      class="modal-dialog w-100 "
    >
      <div
        class="container-content modal-content d-flex justify-content-center align-items-center gap-4 "
      >
        <div
          class="modal-header d-flex justify-content-center"
        >
          <h5
            class="modal-title fs-5"
          />
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-primary button-modal"
            type="button"
          >
            Aceptar
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`rendering pages > loginAdmin > render 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-dark bg-dark"
    >
      <div
        class="container-fluid"
      >
        <div
          class="navbar-brand"
        >
          <span
            class="stretched-link"
          >
            <img
              alt="Boosted - Back to Home"
              height="50"
              loading="lazy"
              src="src/assets/images/logo-orange.svg"
              width="50"
            />
          </span>
          <h1
            class="title"
          >
            Login Administración
          </h1>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="black"
  >
    <div
      class="container py-4 px-3 mx-auto"
    >
      <h4>
        Identifícate
      </h4>
      <form>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="input__name"
          >
            Usuario
          </label>
          <input
            class="form-control"
            id="input__name"
            name="admin_username"
            placeholder="Usuario"
            type="text"
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="input__password"
          >
            Contraseña
          </label>
          <input
            class="form-control"
            id="input__password"
            name="admin_password"
            placeholder="Contraseña"
            type="password"
          />
        </div>
        <div
          class="w-100 text-center"
        >
          <button
            class="btn btn-primary button__login"
            type="submit"
          >
            Entrar
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
